### DISCUSSION TITLE
How to place constraints if my individual is a list of lists?

### DISCUSSION THREAD

**Discussion by Franco-CG** [anyoptimization/pymoo#418] [Q&A]
When my problem was a list  I had no problem definig g1 as:
`g0= (x[1] + 1) + 1*np.sin(x[0]) + 1*np.cos(x[1]) + 1*np.cos(x[0]) + 1*np.cos(6*np.sqrt(x[1]**2 + x[0]**2)) + 1*np.sin(1*np.sqrt(x[1]**2 + x[0]**2))`
`g1=-g0`

So since my individual is now a list of list I thorugh of:
       
```
        constr=([])
        for i in range(len(Rpath)):
          gi=(Rpath[i][1] + 1) + 1*np.sin(Rpath[i][0]) + 1*np.cos(Rpath[i][1]) + 1*np.cos(Rpath[i][0]) + 1*np.cos(6*np.sqrt(Rpath[i][1]**2 + Rpath[i][0]**2)) + 1*np.sin(1*np.sqrt(Rpath[i][1]**2 + Rpath[i][0]**2))
          constr.append(gi)
        g1=[ -x for x in constr]
```
        
        

but then I get the error:
`Exception: ('Problem Error: G can not be set, expected shape (40, 1) but provided (40, 1, 38)', ValueError('cannot reshape array of size 1520 into shape (40,1)'))`

my individual in this case was of len 38

**Comment by blankjul** [comment]
It says that you got the dimensions wrong. It looks like you have a population size of 40 with one constraint, so make sure you set `out["G"]` to a list with 40 values.

**Reply by Franco-CG** [reply]
Thank you for your time. Yes, but I believe my population is 38, I may be wrong (the size is random), but that is what I see when I print the values obtained by my f1.

I found this:
https://github.com/anyoptimization/pymoo/issues/154

I do something similar to this, so I believe my main problem is I dont know to write constraints when is a list of lists.

**Reply by blankjul** [reply]
Check out the getting started guide: https://pymoo.org/getting_started/part_2.html
The problem there has also two constraints.
